cattle/plugins/host_info/memory.py

```diff
@@ -0,0 +1,52 @@
+import platform
+
+
+class MemoryCollector(object):
+    def __init__(self):
+        self.keys = ['memTotal',
+                     'memFree',
+                     'memAvailable',
+                     'buffers',
+                     'cached',
+                     'swapCached',
+                     'active',
+                     'inactive',
+                     'swapTotal',
+                     'swapFree'
+                     ]
+
+        self.unit = 1024.0
+
+    def _get_meminfo_data(self):
+        with open('/proc/meminfo') as f:
+            return f.readlines()
+
+    def _parse_linux_meminfo(self):
+        data = {k: None for k in self.keys}
+        lower_keys = [x.lower() for x in self.keys]
+
+        # /proc/meminfo file has all values in kB
+        mem_data = self._get_meminfo_data()
+        for line in mem_data:
+            line_list = line.split(':')
+
+            try:
+                i = lower_keys.index(line_list[0].lower())
+            except ValueError:
+                i = -1
+
+            if i >= 0:
```

Comment:
I would either put this whole block inside the try and accept the risk of swallowing a subsequent ValueError exception or do a:
```
try:
  # do work
except:
  pass
else:
  # do more work
```